src/transformerbeemcp/rest_api.py

```diff
@@ -0,0 +1,156 @@
+"""REST API for EDIFACT summarization."""
+
+import logging
+import os
+from collections import defaultdict
+from time import time
+
+import jwt
+from fastapi import Depends, FastAPI, HTTPException
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+from jwt import PyJWKClient
+from pydantic import BaseModel
+
+from .summarizer import summarize_edifact
+
+_logger = logging.getLogger(__name__)
+
+# Auth0 configuration
+AUTH0_DOMAIN = os.getenv("AUTH0_DOMAIN", "hochfrequenz.eu.auth0.com")
+AUTH0_AUDIENCE = os.getenv("AUTH0_AUDIENCE", "https://transformer.bee")
+JWKS_URL = f"https://{AUTH0_DOMAIN}/.well-known/jwks.json"
+
+# CORS configuration
+ALLOWED_ORIGINS = os.getenv(
+    "ALLOWED_ORIGINS",
+    "http://localhost:5173,https://nice-mushroom-04ebea203.3.azurestaticapps.net,https://thankful-water-00644131e.3.azurestaticapps.net",
+).split(",")
+
+# Rate limiting: requests per minute per user
+RATE_LIMIT = int(os.getenv("RATE_LIMIT", "10"))
+RATE_WINDOW = int(os.getenv("RATE_WINDOW", "60"))  # seconds
+rate_limit_store: dict[str, list[float]] = defaultdict(list)
+
+app = FastAPI(
+    title="TransformerBee Summarizer",
+    description="REST API for EDIFACT summarization using local LLM",
+    version="0.1.0",
+)
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=ALLOWED_ORIGINS,
+    allow_credentials=True,
+    allow_methods=["POST", "GET", "OPTIONS"],
+    allow_headers=["Authorization", "Content-Type"],
+)
+
+security = HTTPBearer()
+jwks_client: PyJWKClient | None = None
+
+
+def get_jwks_client() -> PyJWKClient:
+    """Lazy initialization of JWKS client."""
+    global jwks_client  # pylint: disable=global-statement
+    if jwks_client is None:
+        jwks_client = PyJWKClient(JWKS_URL)
+    return jwks_client
+
+
+async def verify_token(credentials: HTTPAuthorizationCredentials = Depends(security)) -> dict:
```

Comment:
the return type hint is not precise. feel free to even use literal as dict key or better: use a proper model class.